borsh-derive/src/internals/attributes/item/mod.rs

```diff
@@ -4,36 +4,42 @@ use quote::ToTokens;
 use syn::{spanned::Spanned, Attribute, DeriveInput, Expr, ItemEnum, Path};
 
 pub fn check_item_attributes(derive_input: &DeriveInput) -> Result<(), TokenStream> {
-    for attr in &derive_input.attrs {
-        if attr.path().is_ident(SKIP.0) {
-            return Err(syn::Error::new(
-                derive_input.ident.span(),
-                "`borsh_skip` is not allowed as derive input attribute",
-            )
-            .to_compile_error());
-        }
-        if attr.path().is_ident(BORSH.0) {
-            attr.parse_nested_meta(|meta| {
-                if !meta.path.is_ident(USE_DISCRIMINANT) {
+    // TODO remove in next PR
+    let attr = derive_input.attrs.iter().find(|attr| attr.path() == SKIP);
+
+    if attr.is_some() {
```

Comment:
Just a suggestion, you can use `any` instead of `find` + `is_some`:

```suggestion
    if derive_input.attrs.iter().any(|attr| attr.path() == SKIP) {
```

This might, however make the code split across several lines and actually make it harder to read